sql/sql_show.cc

```diff
@@ -2264,12 +2277,14 @@ class Fill_process_list : public Do_THD_Impl {
     mysql_mutex_unlock(&inspect_thd->LOCK_thd_query);
 
     /* MYSQL_TIME */
-    if (inspect_thd->query_start_in_secs())
-      table->field[5]->store((longlong)(my_micro_time() / 1000000 -
-                                        inspect_thd->query_start_in_secs()),
-                             false);
+    long long time_delta =
+        my_micro_time() -
+        my_timeval_to_micro_time(
+            inspect_thd->query_start_timeval_trunc(DATETIME_MAX_DECIMALS));
+    if (inspect_thd->variables.high_precision_processlist)
+      table->field[5]->store((double)time_delta / 1000000);
     else
-      table->field[5]->store(0, false);
+      table->field[5]->store((long)(time_delta / 1000000), false);
```

Comment:
We have 2 cases it this patch. The first case is `SHOW PROCESSLIST` which calls `mysqld_list_processes()`. It works fine because a result field type is send along with data:
```
field_list.push_back(new Item_return_int("Time", 6, MYSQL_TYPE_DOUBLE));
table->field[5]->store((double)time_delta / 1000000);
```
and
```
field_list.push_back(field = new Item_return_int("Time", 7, MYSQL_TYPE_LONG));
table->field[5]->store((long)(time_delta / 1000000), false);
```
The second case is `SELECT * FROM information_schema.processlist` that calls `Fill_process_list::operator()`.
In this case we send a `double` or a `long` which is not correct according to a definition of the `TIME` filed of `INFORMATION_SCHEMA.PROCESSLIST`.